### PR TITLE
Test all codes in guides/functions

### PR DESCRIPTION
**Pull request by kmaehashi**
Some codes in the document was not tested. Found when reviewing #6179.

### CODE REVIEW

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by hvy** [comment]
LGTM!

**Comment by hvy** [comment]
https://github.com/chainer/chainer/pull/6179 needs to be merged prior to this one.

**Comment by kmaehashi** [comment]
Jenkins, test this please.

File: docs/source/guides/functions.rst [hunk 1]
```diff
@@ -499,6 +525,13 @@ In order to make a convenient module, let's wrap it into a link:
        def forward(self, x):
            return linear(x, self.W, self.b)
 
+.. testcode::
+   :hide:
+
+   x = Variable(np.random.uniform(-1, 1, (2, 2)).astype(np.float32))
+   Linear(2, 2)(x, W, b)
```
Comment (hvy) [inline docs/source/guides/functions.rst]: Should it be the following?
```suggestion
   Linear(2, 2)(x)
```
This is the dump from Jenkins.
```
14:25:09 File "guides/functions.rst", line 1055, in default
14:25:09 Failed example:
14:25:09     x = Variable(np.random.uniform(-1, 1, (2, 2)).astype(np.float32))
14:25:09     Linear(2, 2)(x, W, b)
14:25:09 Exception raised:
14:25:09     Traceback (most recent call last):
14:25:09       File "/usr/lib/python3.5/doctest.py", line 1321, in __run
14:25:09         compileflags, 1), test.globs)
14:25:09       File "<doctest default[0]>", line 2, in <module>
14:25:09         Linear(2, 2)(x, W, b)
14:25:09       File "/work/chainer/chainer/link.py", line 285, in __call__
14:25:09         out = forward(*args, **kwargs)
14:25:09     TypeError: forward() takes 2 positional arguments but 4 were given
```

**Comment by hvy** [comment]
jenkins, test this please.

**Comment by hvy** [comment]
The Jenkins failure seems unrelated.

**Comment by hvy** [comment]
LGTM!

**Pull request merged**